core/services/eth/log_broadcaster.go

```diff
@@ -341,17 +332,38 @@ func (b *logBroadcaster) process(subscription Subscription, chRawLogs <-chan mod
 	}
 }
 
-func (b *logBroadcaster) onRawLog(rawLog models.Log) {
+func (b *logBroadcaster) onRawLog(rawLog types.Log) {
 	for listener := range b.listeners[rawLog.Address] {
 		// Ignore duplicate logs sent back due to reorgs
 		if rawLog.Removed {
 			continue
 		}
 
-		rawLogCopy := rawLog.Copy()
-		lb := logBroadcast{b.orm, &rawLogCopy, listener.JobID()}
-		listener.HandleLog(&lb, nil)
+		// Deep copy the log so that subscribers aren't sharing any state
+		rawLogCopy := copyLog(rawLog)
+		lb := &logBroadcast{log: GethRawLog{rawLogCopy}, orm: b.orm, consumerID: listener.JobID()}
+		listener.HandleLog(lb, nil)
+	}
+}
+
+func copyLog(l types.Log) types.Log {
```

Comment:
Alternatively, you can MarshalJSON + UnmarshalJSON. It's far less efficient but future proof!